src/lib.rs

```diff
@@ -1,58 +1,68 @@
-- **Contexts** represent a certain input state the player can be in, such as "On foot" or "In car". They associate actions with
-  entities and define when those actions are evaluated.
+Inside of this crate, actions are stored as entities with the [`Action<A>`] component, where `A` is your [`InputAction`] type.
+These are associated to contexts via the [`ActionOf<C>`] relationship, where `C` is your context type,
+and can be quickly bound to them using the [`actions!`] macro.
 
-Contexts are regular components. Depending on your type of game, you may have a single global context
-or multiple contexts for different gameplay states. Contexts can be layered, and any number of them can be active at the same time.
-To register a component as an input context, you need to call [`InputContextAppExt::add_input_context`]. By default, contexts are
-evaluated during [`PreUpdate`], but you can customize this by using [`InputContextAppExt::add_input_context_to`] instead.
+The behavior of each action can be further customized using the [`ActionSettings`] component.
 
-Actions are represented by entities with the [`Action<A>`] component, where `A` is a user-defined marker that implements the
-[`InputAction`] trait, which defines [`InputAction::Output`] type - the value the action produces. It could be [`bool`], [`f32`],
-[`Vec2`] or [`Vec3`]. Actions associated with contexts via [`ActionOf`] relationship. We provide the [`actions!`] macro, which is
-similar to [`related!`], but for actions. The relationship is generic over `C` because a single entity can have multiple associated
-contexts.
+## Bindings
 
-Bindings are represented by entities with the [`Binding`] component. It can be constructed from various input types, such as
-[`KeyCode`], [`MouseButton`], [`GamepadAxis`], etc. Bindings associated with actions via [`BindingOf`] relationship. Similar to [`actions!`],
-we provide the [`bindings!`] macro to spawn related bindings. But unlike [`ActionOf<C>`], it's not generic, since each action is represented
-by a separate entity.
+Bindings define how actions are triggered by input sources (e.g. mouse movement or keyboard buttons) that your player might press, like keyboard keys or gamepad buttons.
+We provide support for a variety of input sources out of the box: see the [`Binding`] enum for a full list.
 
+Inside of this crate, bindings are represented by entities with the [`Binding`] component.
+Bindings associated with actions via [`BindingOf`] relationship. Similar to [`actions!`],
+we provide the [`bindings!`] macro to spawn related bindings.
 
 ```
 use bevy::prelude::*;
 use bevy_enhanced_input::prelude::*;
 
-# let mut app = App::new();
-# app.add_plugins(EnhancedInputPlugin);
+#[derive(Component)]
+struct Player;
+
+#[derive(InputAction)]
+#[action_output(bool)]
+struct Jump;
+
+#[derive(InputAction)]
+#[action_output(bool)]
+struct Fire;
+
+let mut app = App::new();
+app.add_plugins(EnhancedInputPlugin);
 app.add_input_context::<Player>();
```

Comment:
The example includes context registration, but the context section comes later. This feels more like an example for "Putting it all together" section. But then it will look like a lot of text before actually showing the code :thinking: 

What do you think about this?
- Move bindings section first and put a very short example where bindings are defined with the mentioned macro. Nothing else, just bindings.
- For actions we showcase how actions are defined for these bindings. No context yet.
- For the context section we wrap it into a context and showcase the spawn and registration.